src/Utilities/ErrorHandling/FormatStacktrace.hpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Formatter for Boost stacktrace, following these docs:
+/// https://www.boost.org/doc/libs/1_78_0/doc/html/stacktrace/getting_started.html#stacktrace.getting_started.global_control_over_stacktrace_o
+
+#pragma once
+
+#include <boost/stacktrace/stacktrace_fwd.hpp>
+
+#include <cstddef>
+#include <ostream>
+#include <string>
+
+namespace stacktrace_detail {
+std::string abbreviated_symbol_name(const std::string& symbol_name);
+std::string get_stack_frame(void* addr);
+#if __has_include(<link.h>)
+std::string addr2line_command(const void* addr);
+#endif
+}  // namespace stacktrace_detail
+
+namespace boost {
+namespace stacktrace {
+
+template <class CharT, class TraitsT, class Allocator>
+std::basic_ostream<CharT, TraitsT>& do_stream_st(
```

Comment:
I find this function name to be quite vague. What does it do? What is `st`? Is this an `impl` or an interface? Since it's being added to the boost namespace, is this naming demanded by boost? If it's the last one, then can you state that, if not can you name is something clearer? Why is this not just in `operator<<`?

I think the answers to these questions should be documented if they cannot be addressed

Reply:
I followed these docs (linked at the top of the file): https://www.boost.org/doc/libs/1_78_0/doc/html/stacktrace/getting_started.html#stacktrace.getting_started.global_control_over_stacktrace_o

However, looks like a plain old `operator<<` works too, so I got rid of the odd naming.